gateway/src/main/java/io/pravega/example/iot/gateway/DataHandler.java

```diff
@@ -68,7 +70,15 @@ public String postData(@Context Request request, String data, @PathParam("remote
 
                 // Write the message to Pravega.
                 Log.debug("routingKey={}, message={}", routingKey, message);
-                final CompletableFuture<Void> writeFuture = Main.getWriter().writeEvent(routingKey, message);
+                // Writing...
+                // Create an immutable writer (in this case using the default settings)
+                final ObjectWriter writer = objectMapper.writer();
```

Comment:
There is no need to create an ObjectWriter. Just call objectMapper.writeValueAsBytes().

Reply:
modified accordingly